resources/problem-vote.js

```diff
@@ -0,0 +1,91 @@
+var voteChart = null;
+
+function init_vote_form() {
+    $('#delete-vote-form').on('submit', function (e) {
+        e.preventDefault();
+        $.ajax({
+            url: $('#delete-vote-form').prop('action'),
+            type: 'POST',
+            data: $('#delete-vote-form').serialize(),
+            success: function () {
+                $('#vote-button').text(gettext('Vote on problem points'));
+                $.featherlight.close();
+            },
+            error: function (data) {
+                alert(interpolate(gettext('Unable to delete vote: %s'), [data.responseText]));
+            }
+        });
+    });
+
+    $('#vote-form').on('submit', function (e) {
+        e.preventDefault();
+        $.ajax({
+            url: $('#vote-form').prop('action'),
+            type: 'POST',
+            data: $('#vote-form').serialize(),
+            success: function (data) {
+                $('#vote-button').text(interpolate(gettext('Change vote (%s)'), [data.points]));
+                $.featherlight.close();
+            },
+            error: function (data) {
+                var errors = data.responseJSON;
+                if (errors === undefined) {
+                    alert(interpolate(gettext('Unable to cast vote: %s'), [data.responseText]));
+                }
+
+                $('#points-error').text(typeof errors === 'object' && 'points' in errors ? errors.points[0] : '');
+                $('#note-error').text(typeof errors === 'object' && 'note' in errors ? errors.note[0] : '');
+            }
+        });
+    });
+}
+
+function reload_vote_graph(data, min_possible_vote, max_possible_vote) {
+    if (voteChart !== null) voteChart.destroy();
+
+    // Give the graph some padding on both sides.
+    var min_points = Math.max(data[0] - 2, min_possible_vote);
+    var max_points = Math.min(data[data.length - 1] + 2, max_possible_vote);
+
+    var xlabels = [];
+    var voteFreq = [];
+    for (var i = min_points; i <= max_points; i++) {
+        xlabels.push(i);
+        voteFreq.push(0);
+    }
+
+    data.forEach(function (x) { voteFreq[x - min_points]++; });
+    var max_number_of_votes = Math.max.apply(null, voteFreq);
+
+    var voteData = {
+        labels: xlabels,
+        datasets: [{
+            label: gettext('Number of votes for this point value'),
+            data: voteFreq,
+            borderColor: 'red',
+            backgroundColor: 'pink',
+        }]
+    };
+    var voteDataConfig = {
+        type: 'bar',
+        data: voteData,
+        options: {
+            responsive: true,
+            scales: {
+                yAxes: [{
+                    ticks: {
+                        precision: 0,
+                        suggestedMax: Math.ceil(max_number_of_votes * 1.2),
+                        beginAtZero: true,
+                    }
+                }],
+                xAxes: [{
+                    ticks: {
+                        beginAtZero: false,
+                    }
+                }]
```

Comment:
Nit, also on following lines

```suggestion
                }],
```